src/main/java/school/hei/haapi/endpoint/rest/controller/AwardedCourseController.java

```diff
@@ -59,4 +64,25 @@ public List<AwardedCourse> getAllAwardedCourseByCriteria(
         .map(mapper::toRest)
         .collect(toList());
   }
+
+  @GetMapping("/teachers/{teacher_id}/awarded_courses")
+  public List<AwardedCourse> getByTeacherId(
+      @PathVariable("teacher_id") String teacherId,
+      @RequestParam(value = "page", defaultValue = "1") PageFromOne page,
+      @RequestParam(value = "page_size", defaultValue = "15") BoundedPageSize pageSize) {
+    User teacher = userService.findById(teacherId);
+    return service.getByTeacherId(teacher.getId(), page, pageSize).stream()
+        .map(mapper::toRest)
+        .collect(toList());
+  }
+
+  @PutMapping("/teachers/{teacher_id}/awarded_courses")
+  public List<AwardedCourse> createOrUpdateAwardedCoursesByTeacherId(
+      @PathVariable("teacher_id") String teacherId,
+      @RequestBody List<CreateAwardedCourse> awardedCourses) {
+    User teacher = userService.findById(teacherId);
+    return service.createOrUpdateAwardedCoursesByTeacherId(teacher.getId(), awardedCourses).stream()
+        .map(mapper::toRest)
+        .collect(toList());
+  }
 }
```

Comment:
U can directly pass the teacherId query param as a paramater for createOrUpdate method to avoid redundancy

Reply:
I used "User teacher = findById(teacherId)" in case the ID doesn't exist, so I can then throw an exception.